Improve NAT logging

MasterMind2k opened the pull request:
Ordinary people usually do not run game from terminal. If an NAT error occurs, user does not see the root cause from logs. Unfortunately, stack traces do not report the issue of failure.

If change is worthy of mentioning, I would add following ChangeLog entry under General improvements:
  * Improve NAT troubleshooting log messages

Log before change:
```
Device found: 192.168.1.1:2828
Type: Upnp
   at Mono.Nat.Upnp.ResponseMessage.Decode(UpnpNatDevice device, String message)
   at Mono.Nat.Upnp.UpnpNatDevice.DecodeMessageFromResponse(Stream s, Int32 length)
   at Mono.Nat.Upnp.UpnpNatDevice.SendMessageAsync(RequestMessage message)
   at Mono.Nat.Upnp.UpnpNatDevice.CreatePortMapAsync(Mapping mapping)
   at Mono.Nat.NatDeviceExtensions.CreatePortMap(INatDevice device, Mapping mapping)
   at OpenRA.Network.Nat.TryForwardPort(Int32 listen, Int32 external) in /home/mastermind/Apps/OpenRA/OpenRA.Game/Network/Nat.cs:line 70
   at Mono.Nat.Upnp.ResponseMessage.Decode(UpnpNatDevice device, String message)
   at Mono.Nat.Upnp.UpnpNatDevice.DecodeMessageFromResponse(Stream s, Int32 length)
   at Mono.Nat.Upnp.UpnpNatDevice.SendMessageAsync(RequestMessage message)
   at Mono.Nat.Upnp.UpnpNatDevice.DeletePortMapAsync(Mapping mapping)
   at Mono.Nat.NatDeviceExtensions.DeletePortMap(INatDevice device, Mapping mapping)
   at OpenRA.Network.Nat.TryRemovePortForward() in /home/mastermind/Apps/OpenRA/OpenRA.Game/Network/Nat.cs:line 89
```

Log after change:
```
Device found: 192.168.1.1:2828
Type: Upnp
Port forwarding failed: Error 725: OnlyPermanentLeasesSupported
   at Mono.Nat.Upnp.ResponseMessage.Decode(UpnpNatDevice device, String message)
   at Mono.Nat.Upnp.UpnpNatDevice.DecodeMessageFromResponse(Stream s, Int32 length)
   at Mono.Nat.Upnp.UpnpNatDevice.SendMessageAsync(RequestMessage message)
   at Mono.Nat.Upnp.UpnpNatDevice.CreatePortMapAsync(Mapping mapping)
   at Mono.Nat.NatDeviceExtensions.CreatePortMap(INatDevice device, Mapping mapping)
   at OpenRA.Network.Nat.TryForwardPort(Int32 listen, Int32 external) in /home/mastermind/Apps/OpenRA/OpenRA.Game/Network/Nat.cs:line 70
Port removal failed: Error 714: NoSuchEntryInArray
   at Mono.Nat.Upnp.ResponseMessage.Decode(UpnpNatDevice device, String message)
   at Mono.Nat.Upnp.UpnpNatDevice.DecodeMessageFromResponse(Stream s, Int32 length)
   at Mono.Nat.Upnp.UpnpNatDevice.SendMessageAsync(RequestMessage message)
   at Mono.Nat.Upnp.UpnpNatDevice.DeletePortMapAsync(Mapping mapping)
   at Mono.Nat.NatDeviceExtensions.DeletePortMap(INatDevice device, Mapping mapping)
   at OpenRA.Network.Nat.TryRemovePortForward() in /home/mastermind/Apps/OpenRA/OpenRA.Game/Network/Nat.cs:line 89
```

MasterMind2k commented:
Pull request improves visibility of underlying NAT problems as reported in #21610

Mailaender commented on OpenRA.Game/Network/Nat.cs:
```diff
@@ -71,7 +71,7 @@ public static bool TryForwardPort(int listen, int external)
 			catch (Exception e)
 			{
-				Console.WriteLine("Port forwarding failed: {0}", e.Message);
+				Log.Write("nat", $"Port forwarding failed: {e.Message}");
 				Log.Write("nat", e.StackTrace);
```
```suggestion
				Log.Write("nat", "Port forwarding failed.");
				Log.Write("nat", e);
```

MasterMind2k committed:
Improve NAT logging

Ordinary people usually do not run game from terminal. If an NAT error
occurs, user does not see the root cause from logs. Unfortunately, stack
traces do not report the issue of failure.

MasterMind2k committed:
Add proposed change

Co-authored-by: Matthias Mailänder <Mailaender@users.noreply.github.com>

Mailaender commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/d9b392fbf9ec64bdd1c5e109c095a09697e95ff4...d380de43882fdf7aef33e9f69a67b6ba03dda69e)
